### ISSUE TITLE
When there is no connection with the server (eg firewall), instead of mentioning a network error, choco says the package was not found

### ISSUE THREAD

**Issue by manad777** [chocolatey/choco#407]
Here's the output of an install or upgrade command on a Windows install with a misconfigured firewall which did not allow the connection to go through.

```
$ cup chocolatey
Chocolatey v0.9.9.8
Upgrading the following packages:
chocolatey
By upgrading you accept licenses for the packages.
chocolatey is not installed. Installing...
chocolatey not installed. The package was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""
 Source(s): "https://chocolatey.org/api/v2/"

Chocolatey upgraded 0/1 package(s). 1 package(s) failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
Failures:
 - chocolatey
```

It would be clearer if choco would output something like "Could not establish connection to server".

**Comment by ferventcoder** [comment]
We may already have a similar issue, but it's likely that it was one that needed migrated from old chocolatey.

**Comment by ferventcoder** [comment]
It's likely something that will need to be fixed first in nuget.core.

**Comment by pyhedgehog** [comment]
It would be even better to be able to configure some number of retry attempts.

**Comment by Mailaender** [comment]
This confused me a bit as it often happens on  @AppVeyor when the VMs have problems: https://ci.appveyor.com/project/OpenRA/openra/build/1.0.7769 which is also confusing to the submitters https://github.com/OpenRA/OpenRA/pull/9927 as one might think it is a install script configuration error not Microsoft Azure or Chocolatey having network problems again.

**Comment by chucker** [comment]
This also happens with e.g. HTTP status code 500. I.e., when something is broken on the server/API end, the client incorrectly assumes the package does not exist.

**Comment by dragon788** [comment]
I've actually run into this one repeatedly when our internal proxy certificate isn't trusted. Attempting an install returns "not installed" per the original post but trying a search returns an ssl/tls error, `The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel.`

**Comment by ferventcoder** [comment]
This may have been fixed with 0.10.4+. It should say that there was a network issue now.
